components/DatePicker.tsx

```diff
@@ -0,0 +1,47 @@
+import { useNasaSearch } from "hooks/useNasaSearch";
+import React, { useState, useRef } from "react";
+import styles from "../styles/DatePicker.module.css";
+
+function DatePicker({ handleDate }: any) {
+  const dateRef = useRef<HTMLInputElement>(null!);
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    try {
```

Comment:
**Comment**

We should remove business logic from the Date Picker and move it to a higher component.

Reply:
Change done!